src/compat/tracker.jl

```diff
@@ -12,14 +12,18 @@ using .Tracker: Tracker,
 using Compat: eachcol
 using LinearAlgebra
 
+# Broadcasting here breaks Tracker
+const TrackedT = Union{TrackedArray, AbstractArray{<:TrackedReal}}
 maporbroadcast(f, x::TrackedArray...) = f.(x...)
-function maporbroadcast(
-    f,
-    x1::TrackedArray{T, N},
-    x::AbstractArray{<:TrackedReal}...,
-) where {T, N}
-    return f.(convert(Array{TrackedReal{T}, N}, x1), x...)
-end
+maporbroadcast(f, x::Union{TrackedArray, AbstractArray{<:TrackedReal}}...) = map(f, x...)
+maporbroadcast(f, x1::TrackedT, x2::AbstractArray) = map(f, x1, x2)
+maporbroadcast(f, x1::AbstractArray, x2::TrackedT) = map(f, x1, x2)
+maporbroadcast(f, x1::TrackedT, x2::AbstractArray, x3::AbstractArray) = map(f, x1, x2, x3)
+maporbroadcast(f, x1::AbstractArray, x2::TrackedT, x3::AbstractArray) = map(f, x1, x2, x3)
+maporbroadcast(f, x1::AbstractArray, x2::AbstractArray, x3::TrackedT) = map(f, x1, x2, x3)
+maporbroadcast(f, x1::TrackedT, x2::TrackedT, x3::AbstractArray) = map(f, x1, x2, x3)
+maporbroadcast(f, x1::AbstractArray, x2::TrackedT, x3::TrackedT) = map(f, x1, x2, x3)
+maporbroadcast(f, x1::TrackedT, x2::AbstractArray, x3::TrackedT) = map(f, x1, x2, x3)
```

Comment:
This does not seem like a robust solution, and needs tests. I still think the easiest solution would be to just change the ReverseDiff part in this PR - since this is what you want to fix and what you added a test for.